lib/datadog/appsec/rate_limiter.rb

```diff
@@ -1,34 +1,15 @@
 # frozen_string_literal: true
 
+require_relative '../core/rate_limiter'
+
 module Datadog
   module AppSec
-    # Simple per-thread rate limiter
-    # Since AppSec marks sampling to keep on a security event, this limits the flood of egress traces involving AppSec
+    # Per-thread rate limiter based on token bucket rate limiter.
+    #
+    # Since AppSec marks sampling to keep on a security event, this limits
+    # the flood of egress traces involving AppSec
     class RateLimiter
-      def initialize(rate)
-        @rate = rate
-        @timestamps = []
-      end
-
-      def limit
-        now = Time.now.to_f
-
-        loop do
-          oldest = @timestamps.first
-
-          break if oldest.nil? || now - oldest < 1
-
-          @timestamps.shift
-        end
-
-        @timestamps << now
-
-        if (count = @timestamps.count) <= @rate
-          yield
-        else
-          Datadog.logger.debug { "Rate limit hit: #{count}/#{@rate} AppSec traces/second" }
-        end
-      end
+      TRACES_KEY = :datadog_security_traces_rate_limiter
 
       class << self
         def limit(name, &block)
```

Comment:
Minor: Should we get rid of `name`? It seems it's been around for a while and we never used it, so I think it may be worth dropping it since we can always re-add when we actually need it?

Reply:
Since I inherit it, I wasn't sure about the whole rewrite because I also don't like an idea of "programming for the future", hence I didn't go full power. If you're saying that it makes sense, then I will shrink the class even more.

Reply:
Yeah... I don't have more context on why this was added initially than you may have, just more of an observation of "this code has been around long enough that we're even refactoring it now, and the future extensibility wasn't needed so far".

Usually I mark as "minor" things I leave up to you the final decision if you want to do anything about it or not, so I can understand if you want to close this one as-is and maybe revisit this later (/never) :)